systems/data-plan/base-rate/test/integration/rate_test.go

```diff
@@ -0,0 +1,59 @@
+package integration
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	pb "github.com/ukama/ukama/systems/data-plan/base-rate/pb/gen"
+	rate "github.com/ukama/ukama/systems/data-plan/package/pkg/db"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/status"
+
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/grpc"
+)
+
+func TestUploadBaseRates(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	// set up connection to server
+	conn, err := grpc.DialContext(ctx, "localhost:9090", grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithBlock())
+
+	if err != nil {
+		logrus.Fatalf("did not connect: %v", err)
+	}
+	defer conn.Close()
+
+	client := pb.NewBaseRatesServiceClient(conn)
+
+	// test data
+	fileURL := "https://raw.githubusercontent.com/ukama/ukama/upload-rates/systems/data-plan/base-rate/template/template.csv"
+	effectiveAt := "2023-03-01T00:00:00Z"
+	simType := rate.ParseType("ukama_data")
```

Comment:
rate.ParseType can you recheck which function is this ?

Reply:
rate.ParseType check that the string passed was a correct simtype.